core/merginuserauth.cpp

```diff
@@ -53,26 +60,50 @@ void MerginUserAuth::setFromJson( QJsonObject docObj )
 
 void MerginUserAuth::saveAuthData()
 {
-  QSettings settings;
-  settings.beginGroup( "Input/" );
-  settings.setValue( "username", mUsername );
-  settings.setValue( "password", mPassword );
-  settings.setValue( "userId", mUserId );
-  settings.setValue( "token", mAuthToken );
-  settings.setValue( "expire", mTokenExpiration );
-  settings.endGroup();
+  if ( !mCredentialStore )
+    return;
+
+  mCredentialStore->writeAuthData( mUsername, mPassword, mUserId, mAuthToken, mTokenExpiration );
 }
 
 void MerginUserAuth::loadAuthData()
 {
-  QSettings settings;
-  settings.beginGroup( QStringLiteral( "Input/" ) );
-  mUsername = settings.value( QStringLiteral( "username" ) ).toString();
-  mPassword = settings.value( QStringLiteral( "password" ) ).toString();
-  mUserId = settings.value( QStringLiteral( "userId" ) ).toInt();
-  mTokenExpiration = settings.value( QStringLiteral( "expire" ) ).toDateTime();
-  mAuthToken = settings.value( QStringLiteral( "token" ) ).toByteArray();
-  settings.endGroup();
+  if ( !mCredentialStore )
+    return;
+
+  connect( mCredentialStore, &CredentialStore::authDataRead, this, [this]
+           ( const QString & username,
+             const QString & password,
+             int userId,
+             const QByteArray & token,
+             const QDateTime & tokenExpiration )
+  {
+    if ( !username.isEmpty() || !password.isEmpty() || userId != -1 )
```

Comment:
I think this is what we want
```suggestion
    if ( !username.isEmpty() && !password.isEmpty() )
```